examples/common/python/connectors/ethereum/contracts/proxy-model/WorkerRegistry.sol

```diff
@@ -0,0 +1,142 @@
+/* Copyright 2019 iExec Blockchain Tech
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+pragma solidity ^0.5.0;
+
+contract WorkerRegistry
+{
+	enum WorkerStatus
+	{
+		NULL,
+		ACTIVE,
+		OFFLINE,
+		DECOMMISSIONED,
+		COMPROMISED
+	}
+
+	struct Worker
+	{
+		uint256   status;
+		uint256   workerType;
+		bytes32   organizationId;
+		bytes32[] appTypeIds;
+		string    details;
+	}
+
+	// WorkerID → Worker
+	mapping(bytes32 => Worker) private m_workers;
+
+	// workerType → organizationId → appTypeId → WorkerID[]
```

Comment:
From #72:
s/appTypeId/appTypeIds/

(I see no appTypeId here, only appTypeIds)

Reply:
`appTypeIds` are arrays of `appTypeId`
to make worker searcheable by invididual appTypeId, we have to split the set of supported appTypeIds and enter as many entries in this multidimensional mapping.

`workerRegister`'s code should show that clearly